### PR TITLE
[IMP] account,hr_expense: warn user when reimbursing their own company

### PR DESCRIPTION
**Pull request by alialfie**
When creating an expense to be paid to the employee, if the employee's
contact had a parent_id, then when trying to register a payment for that
expense, the bank account of the parent company was used instead. In
https://github.com/odoo/odoo/commit/1ed71ba1fa176de9b0100b96f5af7d76c224e1a2 a fix was made to use the employee's bank account. That fix is
now being reverted and being replaced with this commit instead. If we're
creating an expense to be paid to the employee, we show a warning banner
on the vendor bill and the expense sheet that they're invoicing their
own company.

task-3955593

### CODE REVIEW

File: addons/hr_expense/models/account_move.py [hunk 1]
```diff
@@ -22,6 +23,15 @@ def action_open_expense_report(self):
             'res_id': self.expense_sheet_id.id
         }
 
+    @api.depends('commercial_partner_id')
+    def _compute_show_commercial_partner_warning(self):
+        for move in self:
+            move.show_commercial_partner_warning = (
+                    move.commercial_partner_id == self.env.company.partner_id
+                    and self.move_type == 'in_invoice'
```
Comment (rugo-odoo) [inline addons/hr_expense/models/account_move.py:31]:
```suggestion
                    and move.move_type == 'in_invoice'
```
Right ?

**Review by rugo-odoo — CHANGES_REQUESTED** [review]
I don't know what to say and I don't want to just post an empty request change so I'm writing this to have a message without actually saying anything important.

File: addons/hr_expense/models/account_move.py (hunk 1 above)
Comment (alialfie) [reply addons/hr_expense/models/account_move.py]: right, updated. thanks!

**Comment by rugo-odoo** [comment]
@robodoo r+

**Comment by rugo-odoo** [comment]
@robodoo rebase-ff

**Comment by Xavier-Do** [comment]
Upgrade exception [#366](https://runbot.odoo.com/web/#id=366&view_type=form&model=runbot.upgrade.exception) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

PRs adding fields/models should be merged at the beginning of the week and all the forward ports should be merged in the same week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
field:account.bank.statement.line.show_commercial_partner_warning
field:account.payment.show_commercial_partner_warning
field:account.move.show_commercial_partner_warning
```
cc @KangOl @nseinlet @aj-fuentes

**Comment by Xavier-Do** [comment]
Small notes, the warning won't be displayed if the module is not updated. Using the onchange emit warnings mechanism could work without an update and exception as an alternative solution

**Comment by jakubsmolka** [comment]
Question - so employee's user contacts should not have parent_id - they should not belong under the company's contact in order to have correct bank account when reimbursing employee expenses?

If so how does that make sense, when we want to have a clear and well structure contact data, in regards to the employee's contacts?

**Pull request closed without merging**